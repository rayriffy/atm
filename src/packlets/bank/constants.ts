export enum Note {
  Five = '5',
  Ten = '10',
  Twenty = '20',
}

export enum ErrorCode {
  InsufficientFunds = 'InsufficientFunds',
  InsufficientBanknotes = 'InsufficientBanknotes',
  CombinationNotFound = 'CombinationNotFound',
}
