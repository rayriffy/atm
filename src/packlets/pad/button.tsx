import { Button, GridItem, GridItemProps, Heading } from '@chakra-ui/react'
import { Icon } from 'react-iconify-icon-wrapper'

import { ButtonValue } from './constants'

import { usePadLockAtom } from '$context/padLockAtom'

interface Props extends GridItemProps {
  onClick: () => void
  value: ButtonValue
}

export const PinButton = ({ onClick, value, ...rest }: Props) => {
  const lock = usePadLockAtom()

  return (
    <GridItem mx="auto" {...rest}>
      <Button
        rounded="9999px"
        size="lg"
        variant="ghost"
        aspectRatio={1}
        onClick={onClick}
        isDisabled={lock}
      >
        {value !== ButtonValue.Del ? (
          <Heading size="md">{value}</Heading>
        ) : (
          <Icon icon="lucide:delete" width={24} />
        )}
      </Button>
    </GridItem>
  )
}
