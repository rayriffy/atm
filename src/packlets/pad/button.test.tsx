import { render, act } from '@testing-library/react'
import { describe, it, vi, expect } from 'vitest'

import { PadButton } from './button'
import { ButtonValue } from './constants'

const { usePadLockAtomSpy } = vi.hoisted(() => ({
  usePadLockAtomSpy: vi.fn(),
}))
vi.mock('$context/padLockAtom', () => ({
  usePadLockAtom: usePadLockAtomSpy,
}))

describe('<PadButton />', () => {
  it('should match snapshot', () => {
    const { container } = render(<PadButton value={ButtonValue.Zero} />)
    expect(container).toMatchSnapshot()
  })

  it('should be able to click a button', () => {
    const onClickSpy = vi.fn()
    const { getByRole } = render(
      <PadButton value={ButtonValue.Zero} onClick={onClickSpy} />
    )

    expect(onClickSpy).not.toBeCalled()

    act(() => {
      getByRole('button').click()
    })

    expect(onClickSpy).toBeCalledTimes(1)
  })

  it('should disable the button when lock is true', () => {
    usePadLockAtomSpy.mockReturnValue(true)

    const { getByRole } = render(<PadButton value={ButtonValue.Zero} />)

    expect(getByRole('button').attributes).toHaveProperty('disabled')
  })
})
