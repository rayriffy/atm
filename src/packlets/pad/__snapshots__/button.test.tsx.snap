// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`<PadButton /> > should match snapshot 1`] = `
<div>
  <div
    class="css-199gl4c"
  >
    <button
      class="chakra-button css-mw2l4i"
      type="button"
    >
      <h2
        class="chakra-heading css-0"
      >
        0
      </h2>
    </button>
  </div>
</div>
`;
